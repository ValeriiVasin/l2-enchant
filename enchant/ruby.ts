import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const rubyEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.TwoToOne,
  chances: [65, 40, 20, 10],
  items: [Item.Ruby_1, Item.Ruby_2, Item.Ruby_3, Item.Ruby_4, Item.Ruby_5],
});
