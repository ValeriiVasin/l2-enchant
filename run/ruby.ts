import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.Ruby_1, [Item.Ruby_2, Item.Ruby_3, Item.Ruby_4, Item.Ruby_5], 100_000);
