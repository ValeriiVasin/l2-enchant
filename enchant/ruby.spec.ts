import { rubyEnchantMap } from './ruby';

it('enchant map: ruby', () => {
  expect(rubyEnchantMap).toMatchSnapshot();
});
