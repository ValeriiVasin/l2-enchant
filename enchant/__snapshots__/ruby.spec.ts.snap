// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: ruby 1`] = `
Map {
  "Рубин Ур.1" => Object {
    "cost": "Рубин Ур.1",
    "item": "Рубин Ур.1",
    "results": Array [
      Object {
        "chance": 65,
        "item": "Рубин Ур.2",
      },
      Object {
        "chance": 35,
        "item": "Рубин Ур.1",
      },
    ],
  },
  "Рубин Ур.2" => Object {
    "cost": "Рубин Ур.2",
    "item": "Рубин Ур.2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Рубин Ур.3",
      },
      Object {
        "chance": 60,
        "item": "Рубин Ур.2",
      },
    ],
  },
  "Рубин Ур.3" => Object {
    "cost": "Рубин Ур.3",
    "item": "Рубин Ур.3",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Рубин Ур.4",
      },
      Object {
        "chance": 80,
        "item": "Рубин Ур.3",
      },
    ],
  },
  "Рубин Ур.4" => Object {
    "cost": "Рубин Ур.4",
    "item": "Рубин Ур.4",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Рубин Ур.5",
      },
      Object {
        "chance": 90,
        "item": "Рубин Ур.4",
      },
    ],
  },
}
`;
